src/hb.lua

```diff
@@ -38,4 +38,96 @@ function hb.patchBalances(oldBalances)
 	return affectedBalancesAddresses
 end
 
+---@return PrimaryNames|nil affectedPrimaryNamesAddresses
+function hb.createPrimaryNamesPatch()
+	---@type PrimaryNames
+	local affectedPrimaryNamesAddresses = {
+		names = {},
+		owners = {},
+		requests = {},
+	}
+
+	-- if no changes, return early. This will allow downstream code to not send the patch state for this key ('primary-names')
+	if
+		next(_G.HyperbeamSync.primaryNames.names) == nil
+		and next(_G.HyperbeamSync.primaryNames.owners) == nil
+		and next(_G.HyperbeamSync.primaryNames.requests) == nil
+	then
+		return nil
+	end
+
+	-- build the affected primary names addresses table for the patch message
+	for name, _ in pairs(_G.HyperbeamSync.primaryNames.names) do
+		-- we need to send an empty string to remove the name
+		affectedPrimaryNamesAddresses.names[name] = PrimaryNames.names[name] or ""
+	end
+	for owner, _ in pairs(_G.HyperbeamSync.primaryNames.owners) do
+		-- we need to send an empty table to remove the owner primary name data
+		affectedPrimaryNamesAddresses.owners[owner] = PrimaryNames.owners[owner] or {}
+	end
+	for address, _ in pairs(_G.HyperbeamSync.primaryNames.requests) do
+		-- we need to send an empty table to remove the request
+		affectedPrimaryNamesAddresses.requests[address] = PrimaryNames.requests[address] or {}
+	end
+
+	if next(PrimaryNames.names) == nil then
+		affectedPrimaryNamesAddresses.names = {}
+	elseif next(affectedPrimaryNamesAddresses.names) == nil then
+		affectedPrimaryNamesAddresses.names = nil
+	end
+
+	if next(PrimaryNames.owners) == nil then
```

Comment:
Same comments as above for this conditional and the one below as well.

Reply:
answered above